beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -133,6 +186,59 @@ impl<E: EthSpec> RangeBlockComponentsRequest<E> {
         }
     }
 
+    /// Initialize the entries for this component after the column requests have been sent.
+    pub fn initialize_data_columns_from_root_component(
+        &mut self,
+        column_requests: Vec<(DataColumnsByRootRequestId, Vec<ColumnIndex>)>,
+    ) -> Result<(), String> {
```

Comment:
Is it worth creating a proper error type now we have many functions in this struct returning `Result`